### ISSUE TITLE
Reviewers removed from PR after approvals and new commit 

### ISSUE THREAD

**Issue by amix307** [go-gitea/gitea#29649]
### Description

Hi all, I discovered a nasty behaviour when working with Pull Request on Gitea 1.21.7, it is not reproducible on the nightly build on try.gitea.com

So the behaviour is:
1. There is a repository with a configured branch pritection rule for master, more special settings is Dismiss Stale Approvals
2. Create a new branch
3. Commit
4. Create a new Pull Request from the new branch to master
5. After creating Pull Request, assign Reviewers
6. Wait for approve from Reviewers
7. Make a new commit to this Pull Request
8. Those Reviewers who have confirmed approve will be removed from the reviewers list, those who have not confirmed will remain.

Expected behaviour - the approvals should be removed, but not the reviewers

### Gitea Version

1.21.7

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Centos9 Stream

### How are you running Gitea?

selfhosted systemd from binary

### Database

PostgreSQL

**Comment by jpraet** [comment]
See also https://github.com/go-gitea/gitea/pull/28554#issuecomment-1868278043

**Comment by amix307** [comment]
> See also [#28554 (comment)](https://github.com/go-gitea/gitea/pull/28554#issuecomment-1868278043)

For me it's still a slightly different behaviour. I jumped from 1.21.4 to 1.21.7 and there was no such behaviour on the old version

**Comment by amix307** [comment]
Dear developers, please pay attention to this bug, it seems to me that it is very important

**Comment by makar112233** [comment]
We got the same issue on our instase.
Does it seems as "bug", or it is "expected behavior"?

**Comment by jpraet** [comment]
I think there are 2 things:

1. Probably, most users don't want to actually **"dismiss"** stale approvals. That setting used to be implemented as **"ignoring"** stale approvals, but that behavior was changed in Gitea 1.20 to **"dismiss"** them. In Gitea 1.22, there is a new branch protection setting "ignore stale approvals" that restores the old behavior: https://github.com/go-gitea/gitea/pull/28498
2. When a review is dismissed, should it revert to the "review requested" state or not? See https://github.com/go-gitea/gitea/pull/28554#issuecomment-1868278043

**Comment by jpraet** [comment]
Isn't that the exact same problem I had pointed out in https://github.com/go-gitea/gitea/pull/28554#issuecomment-1868278043?

The yellow dot is the indication of a review request:

![image](https://github.com/go-gitea/gitea/assets/2212615/061b78ce-91c3-47e5-9922-52a0bf156dcc)

Previously (< 1.21.6), when a review is submitted, and later dismissed because of new commits, the yellow dot would be displayed again.

As of https://github.com/go-gitea/gitea/pull/28554 review requests get deleted when submitting a review. So when the submitted review is dismissed, there is no indication in the UI anymore.

**Comment by amix307** [comment]
Yeah, I guess you're right, I was inattentive. Do you think this is planned to be fixed or somehow given the ability to control it?

**Comment by yp05327** [comment]
If you remove the rule, and do the same thing, you will see an `x` button, and the description of it is `dismiss the reviewer`.
If you click it, the reviewer will be removed. So as the description of this option:
> When new commits that change the content of the pull request are pushed to the branch, old approvals will be dismissed.

`dismissed` means `removed`.

Maybe we need another option which just re-request the review request.
